src/app/organizations/organization-members-edit/organization-members-edit.component.ts

```diff
@@ -50,41 +68,77 @@ export class OrganizationMembersEditComponent implements OnInit {
     const emptyFormControl = new FormControl();
     emptyFormControl.valueChanges.pipe(take(1)).subscribe(() => {
       emptyFormControl.addValidators(Validators.required);
+      emptyRoleDropdown.setValue(this.newUserDefaultRoles);
       this.addEmptyFormControl();
     });
-    this.userRefs.push(emptyFormControl);
+    const emptyRoleDropdown = new FormControl([]);
+    const emptyFormGroup = new FormGroup({
+      userName: emptyFormControl,
+      selectedRoles: emptyRoleDropdown,
+    });
+    this.userRefs.push(emptyFormGroup);
+  }
+
+  mapRolesToUsers(): Record<string, string[]> {
+    const userRoles: Record<string, string[]> = {};
+    this.roleBindings.items.forEach((item) => {
+      item.subjects.forEach((subj) => {
+        if (!userRoles[subj.name]) userRoles[subj.name] = [];
+        userRoles[subj.name].push(item.roleRef.name);
+      });
+    });
+    return userRoles;
   }
 
   save(): void {
-    const userRefs = this.form.value.userRefs as string[];
-    userRefs.pop();
-    this.kubernetesClientService
-      .updateOrganizationMembers({
+    const userNames: string[] = this.form.value.userRefs
+      .map((userDetails: { userName: string; selectedRoles: string[] }) => userDetails.userName)
+      .filter((val: string) => val);
+
+    const rolesToSubjects: Record<string, string[]> = {};
+    this.form.value.userRefs.forEach((userDetails: { userName: string; selectedRoles: string[] }) => {
+      userDetails.selectedRoles?.forEach((role) => {
+        if (!rolesToSubjects[role]) rolesToSubjects[role] = [];
```

Comment:
Can you add curly braces.

Reply:
Done, I've also added the corresponding eslint rule